test/burn/org/apache/cassandra/net/ConnectionBurnTest.java

```diff
@@ -467,26 +491,42 @@ public void run() throws ExecutionException, InterruptedException, NoSuchFieldEx
                     }
                 });
 
-                while (deadline > nanoTime() && failed.getCount() > 0)
+                int added = 0;
```

Comment:
at most we need a boolean, but I'd say this should be a part of the loop guard; perhaps use a do/while loop and simply check 0 == Queues.drainUninterruptibly()

Reply:
Good suggestion, fixed.